### DISCUSSION TITLE
Dataset size in "Power BI Premium Capacity Metrics" vs DAX Studio

### DISCUSSION THREAD

**Discussion by ImranDJCS** [DaxStudio/DaxStudio#707] [Q&A] [answered]
Hi there,
I created a dataset and can see its size **3780.52**MB in "_Power BI Premium Capacity Metrics_". But DAX Studio is showing the size of **5.60MB** for the same dataset. 
Could you advise why two different sizes for the same dataset?
Thank you

![image](https://user-images.githubusercontent.com/83199486/132194486-501538c2-665a-4247-9243-aa7e4c07cc87.png)

![image](https://user-images.githubusercontent.com/83199486/132194118-bad73159-9361-4788-abff-39805dd64113.png)

**Comment by dgosbell** [comment, marked as answer]
Did you connect DAX Studio to the XMLA Endpoint and connect to the same dataset in the same workspace?

**Reply by ImranDJCS** [reply]
Did check this morning post refresh my "Power BI Premium Capacity Metrics" App. Also generated new metrics from DAX Studio. Now things look better than before.
Power BI shows **33.91MB** and DAX Studio shows **5.59MB**.
Is it a legitimate size difference when using two different tools to size the same dataset?

**Reply by dgosbell** [reply]
So I can tell you exactly which DMV queries we use to calculate the model size, but I don't believe the metrics app has a breakdown of what is using the memory. It's possible that the metrics app is including session cache and possibly even mashup engine instances. I'll see if I can find out any information on what the metrics app is including in its figures.

**Reply by ImranDJCS** [reply]
Sure. Any details will be appreciated which can help to understand the difference.
Thank you

**Reply by dgosbell** [reply]
OK, I had a chat to someone at Msft. DAX Studio purely reports the memory used by your data model. While in the Premium Metrics App it also includes any other memory used to run your data model so there will be non-Vertipaq metadata, sessions, global scopes, RLS, etc

**Reply by ImranDJCS** [reply]
Thank you Darren.
Appreciate that.

**Reply by pawarbi** [reply]
Hi @dgosbell can you please share which DMVs are used to calculate the overall size of the model as reported in Summary? Thanks.

**Reply by dgosbell** [reply]
@pawarbi we use a combination of  $SYSTEM.DISCOVER_STORAGE_TABLE_COLUMNS and $SYSTEM.DISCOVER_STORAGE_TABLE_COLUMN_SEGMENTS to get the detailed usage and then add up all the details to calculate the total. This class has all the different calls we use https://github.com/sql-bi/VertiPaq-Analyzer/blob/master/src/Dax.Model.Extractor/DmvExtractor.cs
